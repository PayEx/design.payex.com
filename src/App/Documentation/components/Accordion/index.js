import React, { useEffect } from "react";

import { ComponentPreview, DocContainer } from "#";
import accordion from "$/px-script/main/accordion";
import AccordionComponent from "@/Accordion";

const items = [
    {
        title: "Foo",
        content: <p>My accordion content</p>
    },
    {
        title: "Bar",
        content: <p>My accordion content</p>
    },
    {
        title: "Baz",
        content: <>
                <p>My accordion content</p>
                <p>My accordion content</p>
                <p>My accordion content</p>
                <p>My accordion content</p>
                <p>My accordion content</p>
            </>
    }
];

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>Accordion.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <AccordionComponent items={items} />
        </ComponentPreview>
    </>
);

const Accordion = () => {
    useEffect(() => accordion.init());

    return (
        <DocContainer docToc>
            <p className="lead">
                Accordion
            </p>
            <Overview />
        </DocContainer>
    );
};

export default Accordion;

/* For testing */
export { Overview };
