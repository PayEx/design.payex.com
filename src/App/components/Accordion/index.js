import React from "react";

const Accordion = ({ items }) => (
    <div className="accordion">
        {items ? items.map(({ title, content }, i) => (
            <div key={i} className="accordion-item">
                <h5>{title}</h5>
                <div className="accordion-content">
                    {content}
                </div>
            </div>
        )) : null}
    </div>
);

export default Accordion;
