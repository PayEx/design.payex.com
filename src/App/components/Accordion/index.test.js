import React from "react";
import { shallow } from "enzyme";

import Accordion from "./index";

const item = [
    {
        title: "Foo",
        content: <p>My accordion content</p>
    }
];

const items = [
    {
        title: "Foo",
        content: <p>My accordion content</p>
    },
    {
        title: "Bar",
        content: <p>My accordion content</p>
    },
    {
        title: "Baz",
        content: <p>My accordion content</p>
    }
];

describe("Component: Accordion", () => {
    it("is defined", () => {
        expect(Accordion).toBeDefined();
    });

    it("returns an empty accordion container if the items props is not provided", () => {
        const wrapper = shallow(<Accordion />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<div className="accordion"></div>)).toEqual(true);
    });

    it("returns an accordion containing the item", () => {
        const wrapper = shallow(<Accordion items={item}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(items[0].title)).toEqual(true);
        expect(wrapper.contains(<div className="accordion-content">{items[0].content}</div>)).toEqual(true);
    });

    it("returns an accordion with several accordion items", () => {
        const wrapper = shallow(<Accordion items={items}/>);

        expect(wrapper).toMatchSnapshot();
        items.forEach(item => {
            expect(wrapper.contains(item.title)).toEqual(true);
            expect(wrapper.contains(<div className="accordion-content">{item.content}</div>)).toEqual(true);
        });
    });
});
