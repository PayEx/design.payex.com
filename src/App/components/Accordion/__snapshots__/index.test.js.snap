// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Accordion returns an accordion containing the item 1`] = `
<div
  className="accordion"
>
  <div
    className="accordion-item"
    key="0"
  >
    <h5>
      Foo
    </h5>
    <div
      className="accordion-content"
    >
      <p>
        My accordion content
      </p>
    </div>
  </div>
</div>
`;

exports[`Component: Accordion returns an accordion with several accordion items 1`] = `
<div
  className="accordion"
>
  <div
    className="accordion-item"
    key="0"
  >
    <h5>
      Foo
    </h5>
    <div
      className="accordion-content"
    >
      <p>
        My accordion content
      </p>
    </div>
  </div>
  <div
    className="accordion-item"
    key="1"
  >
    <h5>
      Bar
    </h5>
    <div
      className="accordion-content"
    >
      <p>
        My accordion content
      </p>
    </div>
  </div>
  <div
    className="accordion-item"
    key="2"
  >
    <h5>
      Baz
    </h5>
    <div
      className="accordion-content"
    >
      <p>
        My accordion content
      </p>
    </div>
  </div>
</div>
`;

exports[`Component: Accordion returns an empty accordion container if the items props is not provided 1`] = `
<div
  className="accordion"
/>
`;
